exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/transactions/ThreadLocalTransactionManager.kt

```diff
@@ -23,22 +23,40 @@ class ThreadLocalTransactionManager(
     private val setupTxConnection: ((ExposedConnection<*>, TransactionInterface) -> Unit)? = null
 ) : TransactionManager {
     @Volatile
-    override var defaultRepetitionAttempts: Int = db.config.defaultRepetitionAttempts
-        @Deprecated("Use DatabaseConfig to define the defaultRepetitionAttempts")
-        @TestOnly
-        set
+    override var defaultMaxAttempts: Int = db.config.defaultMaxAttempts
 
     @Volatile
-    override var defaultMinRepetitionDelay: Long = db.config.defaultMinRepetitionDelay
-        @Deprecated("Use DatabaseConfig to define the defaultMinRepetitionDelay")
-        @TestOnly
-        set
+    override var defaultMinRetryDelay: Long = db.config.defaultMinRetryDelay
 
     @Volatile
-    override var defaultMaxRepetitionDelay: Long = db.config.defaultMaxRepetitionDelay
-        @Deprecated("Use DatabaseConfig to define the defaultMaxRepetitionDelay")
-        @TestOnly
-        set
+    override var defaultMaxRetryDelay: Long = db.config.defaultMaxRetryDelay
```

Comment:
All the older properties in this class have a deprecated setter to discourage writing values. I haven't added this to the new properties because I hoped there might be a better way to do this.

The 2 ways to set these properties are:
```kt
Database.connect(
     "jdbc:h2:mem:db1;DB_CLOSE_DELAY=-1;", "org.h2.Driver",
     databaseConfig = DatabaseConfig { },
     // this setter would show a deprecation warning if the annotation was added
     manager = { ThreadLocalTransactionManager(it).apply { defaultMaxAttempts = 1 } }
)

transaction {
    // this would not show a deprecation warning because it uses the interface property
    db.transactionManager.defaultMaxAttempts = 1
}
```